File: src/my_cpp_pkg/src/hw_status_publisher.cpp
#include "my_robot_interfaces/msg/hardware_status.hpp"
#include "rclcpp/rclcpp.hpp"

using my_robot_interfaces::msg::HardwareStatus;

class HardwareStatusPublisher : public rclcpp::Node {
public:
  HardwareStatusPublisher() : Node("robot_news_station"), robot_name_("R2D2") {
    publisher_ = this->create_publisher<HardwareStatus>("hardware_status", 10);

    timer_ = this->create_wall_timer(
        std::chrono::milliseconds(500),
        std::bind(&HardwareStatusPublisher::publishHardwareStatus, this));
    RCLCPP_INFO(this->get_logger(), "Robot News has been started");
  }

private:
  void publishHardwareStatus() {
    auto msg = HardwareStatus();
    msg.temperature = 57;
    msg.are_motors_ready = true;
    msg.debug_message = "Motors are too hot!";
    publisher_->publish(msg);
  }

  std::string robot_name_;
  rclcpp::Publisher<HardwareStatus>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char const *argv[]) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<HardwareStatusPublisher>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: src/my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3
"""
This is a simple node that publishes a string message to the topic "my_first_topic"
"""  # noqa


import rclpy
from rclpy.node import Node

from my_robot_interfaces.msg import HardwareStatus


class HardwareStatusPublisher(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")
        self.hw_status_publisher_ = self.create_publisher(
            HardwareStatus, "hardware_status", 10
        )
        self.timer_ = self.create_timer(1.0, self.publish_hw_status)
        self.get_logger().info("Hardware status publisher has been started.")

    def publish_hw_status(self):
        msg = HardwareStatus()
        msg.temperature = 45
        msg.are_motors_ready = True
        msg.debug_message = "Nothing special here"
        self.hw_status_publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisher()
    rclpy.spin(node)  # mantém o programa rodando
    rclpy.shutdown()


if __name__ == "__main__":
    main()
